filter.go

```diff
@@ -82,7 +82,10 @@ func CompileFilter(filter string) (*ber.Packet, error) {
 	if err != nil {
 		return nil, err
 	}
-	if pos != len(filter) {
```

Comment:
I see how this improves the error message, but I'm surprised this fixes a panic...